### PR TITLE
Only copy if cell is present when expand_merged_ranges: true

### PR DESCRIPTION
**Pull request by ttilberg**
This fixes https://github.com/roo-rb/roo/issues/506 where `expand_merged_ranges: true` was shown to create `nil` in the `@cells` hash. This then would cause other issues where cell objects were expected to exist (such as during `clean: true`).

### CODE REVIEW

**Comment by ttilberg** [comment]
Hello, I see that there are tests available for `expand_merged_ranges` at `test/roo/test_excelx.rb:101` -- but I'm hesitant to later `test/files/merged_ranges.xls` as I don't have Excel (only LibreOffice).

Is this okay? I'd like to figure out how to add a test for this to assert that we don't copy nil cells.

Or should I just add the example file from the issue?

**Comment by ttilberg** [comment]
I haven't looked at this, but if `doc.xpath('/worksheet/mergeCells/mergeCell')` is able to filter to nodes with content, this would prevent us from doing the work at all. I'll check this when I get back to my desk.

**Comment by ttilberg** [comment]
I just looked into filter during node selection, but there does not appear to be any difference:

```
[3] pry(#<Roo::Excelx::SheetDoc>)> doc.xpath('/worksheet/mergeCells').to_xml
=> "<mergeCells count=\"2\">\n  <mergeCell ref=\"B3:B4\"/>\n  <mergeCell ref=\"B5:B6\"/>\n</mergeCells>"
```

**Comment by ttilberg** [comment]
> Hello, I see that there are tests available for `expand_merged_ranges` at `test/roo/test_excelx.rb:101` -- but I'm hesitant to alter `test/files/merged_ranges.xls` as I don't have Excel (only LibreOffice).
> 
> Is this okay? I'd like to figure out how to add a test for this to assert that we don't copy nil cells.
> 
> Or should I just add the example file from the issue?

I just pushed a commit with a test for this condition, which included me editing `test/files/merged_ranges.xls` in LibreOffice to include a merged range without content. I want to be explicit about this change to this file in case it is not appropriate to have edited it with LibreOffice (or at all). If you wish a separate file is created, I'd be happy to send that instead, but it felt unnecessary.

**Comment by ttilberg** [comment]
@chopraanmol1 would you mind taking a look at this? I was asked by a coworker if this patch was accepted yet and I noticed this hasn't been looked at yet. I'd love to make whatever changes necessary to get this accepted. Thanks!

File: lib/roo/excelx/sheet_doc.rb [hunk 1]
```diff
@@ -207,7 +207,7 @@ def expand_merged_ranges(cells)
         end
         # Duplicate value into all cells in merged range
         merges.each do |dst, src|
-          cells[dst] = cells[src]
+          cells[dst] = cells[src] if cells[src]
```
Comment (chopraanmol1) [inline lib/roo/excelx/sheet_doc.rb]: I think we should add this check before adding  range to merges hash. Just after https://github.com/ttilberg/roo/blob/e46239acbde159ac1b63de681450a54ffbbb31e2/lib/roo/excelx/sheet_doc.rb#L200 seems like right place to add this check

File: test/roo/test_excelx.rb [hunk 2]
```diff
@@ -133,6 +133,13 @@ def test_expand_merged_range
     end
   end
 
+  def test_expand_merged_range_doesnt_insert_nil_values
+    options = { expand_merged_ranges: true }
+    xlsx = roo_class.new(File.join(TESTDIR, "merged_ranges.xlsx"), options)
+
+    refute_includes xlsx.sheet_for(0).cells.values, nil, "`nil` was copied into the cells hash from an empty merged range"
+  end
```
Comment (chopraanmol1) [inline test/roo/test_excelx.rb]: Can you also write a test case which verifies that issue posted in https://github.com/roo-rb/roo/issues/506 is actually solved

Comment (ttilberg) [reply test/roo/test_excelx.rb]: Sure thing.

I originally added the same problem to the existing test file so I didn't clutter up your project. Would you like me to add the issue's `test.xlsx` file and write a test against that file specifically?

If yes, do you want me to revert the changes to the existing `merged_ranges.xlsx` test file, or keep both tests?

File: lib/roo/excelx/sheet_doc.rb (hunk 1 above)
Comment (ttilberg) [reply lib/roo/excelx/sheet_doc.rb]: Good idea. I was having a hard time parsing the above code (the `tl` `br`), but you're right. If `cells[tl]` is nil, then none of it's iterations should get collected, potentially saving some work. I updated the variable names to help make a bit more clear what was going on -- I didn't see that `tl` and `br` were commonly used elsewhere. If these have more context than I know of, I'm happy to change them back.

Thanks!

**Comment by ttilberg** [comment]
@chopraanmol1 Can you please review this, or ping someone else that might be appropriate to review and accept this PR? Thank you for your time.
